armi/bookkeeping/newReportUtils.py

```diff
@@ -0,0 +1,711 @@
+            duct = b.getComponent(Flags.DUCT)
+            clad = b.getComponent(Flags.CLAD)
+            wire = b.getComponent(Flags.WIRE)
+            designInfo["hot sd"].append(b.getSmearDensity(cold=False))
+            designInfo["sd"].append(b.getSmearDensity())
+            designInfo["ductThick"].append(
+                (duct.getDimension("op") - duct.getDimension("ip")) * 5.0
+            )  # convert to mm and divide by 2
+            designInfo["cladThick"].append(
+                (clad.getDimension("od") - clad.getDimension("id")) * 5.0
+            )
+            pinOD = clad.getDimension("od") * 10.0
+            wireOD = wire.getDimension("od") * 10.0
+            pitch = pinOD + wireOD  # pitch has half a wire on each side.
+            assemPitch = b.getPitch() * 10  # convert cm to mm.
+            designInfo["pinOD"].append(pinOD)
+            designInfo["wireOD"].append(wireOD)
+            designInfo["pin pitch"].append(pitch)
+            pinToDuctGap = b.getPinToDuctGap()
+            if pinToDuctGap is not None:
+                designInfo["pinToDuct"].append(b.getPinToDuctGap() * 10.0)
+            designInfo["assemPitch"].append(assemPitch)
+            designInfo["duct gap"].append(assemPitch - duct.getDimension("op") * 10.0)
+            designInfo["nPins"].append(b.p.nPins)
+            designInfo["zrFrac"].append(fuel.getMassFrac("ZR"))
+
+        # assumption made that all lists contain only numerical data
+        designInfo = {key: numpy.average(data) for key, data in designInfo.items()}
+        tableRows = newReports.Table("Pin Design", header=None)
+        dimensionless = {"sd", "hot sd", "zrFrac", "nPins"}
+        for key, average_value in designInfo.items():
+            dim = "{0:10s}".format(key)
+            val = "{0:.4f}".format(average_value)
+            if key not in dimensionless:
+                val += " mm"
+            # want to basically add a row to a table with dim ---> val
+            tableRows.addRow([dim, val])
+
+        a = core.refAssem
+        tableRows.addRow(
+            ["Fuel Height (cm):", "{0:.2f}".format(a.getHeight(Flags.FUEL))]
+        )
+        tableRows.addRow(
+            ["Plenum Height (cm):", "{0:.2f}".format(a.getHeight(Flags.PLENUM))]
+        )
+
+        return tableRows
+    except Exception as error:  # pylint: disable=broad-except
+        runLog.warning("Pin summarization failed to work")
+        runLog.warning(error)
+
+
+def insertAreaFractionsReport(block, report):
+    """Adds to an Assembly Area Fractions table subsection of the Comprehensive Section
+    of the report.
+
+        Parameters
+        ----------
+        block: Block
+        report: ReportContent
+
+    """
+
+    for c, frac in block.getVolumeFractions():
+
+        report[COMPREHENSIVE_REPORT][ASSEMBLY_AREA].addRow(
+            [c.getName(), "{0:10f}".format(c.getArea()), "{0:10f}".format(frac)]
+        )
+
+
+def setDimensionReport(comp):
```

Comment:
If this returns a Table, lets call it, `get`, or `create`, rather than `set`. `set` implies side effects.